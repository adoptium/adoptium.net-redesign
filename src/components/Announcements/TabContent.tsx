import React from "react"
import { Link } from "../Link"

const TabContent = ({ posts }) => {
  return (
    <article>
      {posts.map((post, index) => (
        <>
          <Link to={post.node.fields.postPath} key={index}>
          <div className="text-[#c4bfce">
            <p className="texxt-[12px] leading-[133.333%] text-lightgrey mb-0">
              {Math.floor((new Date().getTime() - new Date(post.node.frontmatter.date).getTime()) / (1000 * 60 * 60 * 24 * 7))} weeks ago
            </p>

            <h3 className="text-[20px] leading-[140%] text-white">
              {post.node.frontmatter.title}
            </h3>
            <p className="tab-button-text text-sm text-lightgrey pt-2">
              {post.node.frontmatter.description}
            </p>
          </div>
          </Link>
          <span className="h-[1px] w-full bg-[#3E3355] inline-block my-4"></span>
        </>
      ))}
    </article>
  )
}

export default TabContent
