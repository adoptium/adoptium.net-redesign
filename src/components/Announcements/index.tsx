import React, { useState } from "react"
import { useStaticQuery, graphql } from "gatsby"
import Sidebar from "../Common/Sidebar"
import TabContent from "./TabContent"

const Announcements = ({ handleClose }) => {
  const data = useStaticQuery(graphql`
    query {
      allMdx(sort: {frontmatter: {date: DESC}}) {
        edges {
          node {
            id
            frontmatter {
              title
              date
              description
              tags
            }
            fields {
              postPath
            }
          }
        }
      }
    }
  `)

  const latest_posts = data.allMdx.edges.slice(0, 4)
  const latest_releases = data.allMdx.edges.filter(
    ({ node }) => node.frontmatter.tags.includes("release-notes")
  )

  const [active, setActive] = useState("Updates")
  return (
    <Sidebar onClose={handleClose} header="Announcements">
      <div className="flex items-center gap-12 relative">
        <span className="h-[1px] w-full bg-[#3E3355] inline-block absolute bottom-[0.7px] z-[-1]"></span>
        <div className="flex space-x-10 whitespace-nowrap   py-3">
          <button onClick={() => setActive("Updates")}>
            <span
              className={`py-3 w-full tab-button-text
                outline-hidden cursor-pointer transition-all duration-200 ease-in-out ${active === "Updates" ? "border-primary border-b-2 text-white" : "text-[#8a809e] border-transparent border-b"}`}
            >
              Updates
            </span>
          </button>
          {/* <button onClick={() => setActive("Events")}>
            <span
              className={`py-3 w-full tab-button-text
                outline-hidden cursor-pointer transition-all duration-200 ease-in-out ${active === "Events" ? "border-primary border-b-2 text-white" : "text-[#8a809e] border-transparent border-b"}`}
            >
              Events
            </span>
          </button> */}
          <button onClick={() => setActive("Releases")}>
            <span
              className={`py-3 w-full tab-button-text
                outline-hidden cursor-pointer transition-all duration-200 ease-in-out ${active === "Releases" ? "border-primary border-b-2 text-white" : "text-[#8a809e] border-transparent border-b"}`}
            >
              Releases
            </span>
          </button>
        </div>
      </div>
      <div className="mt-6 grow overflow-hidden h-full pb-28">
        <div className="overflow-auto h-[88%] scroll-sidebar">
          {active === "Updates" && <TabContent posts={latest_posts} />}
          {/* {active === "Events" && <TabContent posts={latest_posts} />} */}
          {active === "Releases" && <TabContent posts={latest_releases} />}
        </div>
      </div>
    </Sidebar>
  )
}

export default Announcements
