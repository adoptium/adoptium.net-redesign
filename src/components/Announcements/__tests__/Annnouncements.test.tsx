import React from "react"
import { render } from "@testing-library/react"
import { describe, expect, it, vi } from "vitest"
import Announcements from ".."

const mockData = ({id, days}) => ({
  node: {
    id,
    frontmatter: {
      title: `Test Announcement ${id}`,
      date: Date.now() - 1000 * 60 * 60 * 24 * days,
      description: `Description for announcement ${id}`,
      tags: ["release-notes"],
    },
    fields: {
      postPath: `/test-announcement-${id}`,
    },
  },
})

vi.mock("gatsby", async () => {
  const actualGatsby = await vi.importActual("gatsby")
  return {
    ...actualGatsby,
    // This no-op function prevents the error by simply returning the query string.
    graphql: (strings: TemplateStringsArray, ..._args: any[]) => strings[0],
    // Provide your test data for useStaticQuery.
    useStaticQuery: () => ({
      allMdx: {
        edges: [
          mockData({id: 1, days: 1}), // 1 day ago
          mockData({id: 2, days: 3}), // 3 days ago
          mockData({id: 3, days: 7}), // 1 week ago
          mockData({id: 4, days: 60}),  // 8 weeks ago
        ],
      },
    }),
  }
})

describe("Announcements component", () => {
  it("Announcements renders correctly", () => {
    const handleCloseMock = vi.fn()
    const { container } = render(
      <Announcements handleClose={handleCloseMock} />,
    )
    expect(container).toMatchSnapshot()
  })
})
