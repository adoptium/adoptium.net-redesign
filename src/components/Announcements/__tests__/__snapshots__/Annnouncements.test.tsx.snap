// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Announcements component > Announcements renders correctly 1`] = `
<div>
  <div
    class="z-[100] fixed top-0 right-0 w-full max-w-[480px] bg-[#0e002a] p-6 pt-8 h-screen rounded-s-3xl"
  >
    <div
      class="flex items-center justify-between"
    >
      <h1
        class="text-2xl font-semibold"
      >
        Announcements
      </h1>
      <button
        class="-m-2.5 rounded-md p-2.5 text-white-700"
        type="button"
      >
        <span
          class="sr-only"
        >
          Close menu
        </span>
        <div
          class="border-2 border-[#3e3355] p-3 rounded-full"
        >
          <svg
            class="text-xl"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2.146 2.854a.5.5 0 1 1 .708-.708L8 7.293l5.146-5.147a.5.5 0 0 1 .708.708L8.707 8l5.147 5.146a.5.5 0 0 1-.708.708L8 8.707l-5.146 5.147a.5.5 0 0 1-.708-.708L7.293 8z"
            />
          </svg>
        </div>
      </button>
    </div>
    <div
      class="flex items-center gap-12 relative"
    >
      <span
        class="h-[1px] w-full bg-[#3E3355] inline-block absolute bottom-[0.7px] z-[-1]"
      />
      <div
        class="flex space-x-10 whitespace-nowrap   py-3"
      >
        <button>
          <span
            class="py-3 w-full tab-button-text
                outline-none cursor-pointer transition-all duration-200 ease-in-out border-primary border-b-2 text-white"
          >
            Updates
          </span>
        </button>
        <button>
          <span
            class="py-3 w-full tab-button-text
                outline-none cursor-pointer transition-all duration-200 ease-in-out text-[#8a809e] border-transparent border-b"
          >
            Releases
          </span>
        </button>
      </div>
    </div>
    <div
      class="mt-6 grow overflow-hidden h-full pb-28"
    >
      <div
        class="overflow-auto h-[88%] scroll-sidebar"
      >
        <article>
          <a>
            <div
              class="text-[#c4bfce"
            >
              <p
                class="texxt-[12px] leading-[133.333%] text-lightgrey mb-0"
              >
                213
                 weeks ago
              </p>
              <h3
                class="text-[20px] leading-[140%] text-white"
              >
                Mock Title 1
              </h3>
              <p
                class="tab-button-text text-sm text-lightgrey pt-2"
              >
                Mock Description 1
              </p>
            </div>
          </a>
          <span
            class="h-[1px] w-full bg-[#3E3355] inline-block my-4"
          />
          <a>
            <div
              class="text-[#c4bfce"
            >
              <p
                class="texxt-[12px] leading-[133.333%] text-lightgrey mb-0"
              >
                213
                 weeks ago
              </p>
              <h3
                class="text-[20px] leading-[140%] text-white"
              >
                Mock Title 2
              </h3>
              <p
                class="tab-button-text text-sm text-lightgrey pt-2"
              >
                Mock Description 2
              </p>
            </div>
          </a>
          <span
            class="h-[1px] w-full bg-[#3E3355] inline-block my-4"
          />
          <a>
            <div
              class="text-[#c4bfce"
            >
              <p
                class="texxt-[12px] leading-[133.333%] text-lightgrey mb-0"
              >
                213
                 weeks ago
              </p>
              <h3
                class="text-[20px] leading-[140%] text-white"
              >
                Mock Title 3
              </h3>
              <p
                class="tab-button-text text-sm text-lightgrey pt-2"
              >
                Mock Description 3
              </p>
            </div>
          </a>
          <span
            class="h-[1px] w-full bg-[#3E3355] inline-block my-4"
          />
        </article>
      </div>
    </div>
  </div>
</div>
`;
